lib/msf/core/opt_base.rb

```diff
@@ -22,15 +22,39 @@ class OptBase
     # attrs[3] = possible enum values
     # attrs[4] = Regex to validate the option
     #
-    def initialize(in_name, attrs = [], aliases: [])
+    # Attrs can also be specified explicitly via named parameters, or attrs can
+    # also be a string as standin for the required description field.
+    #
+    def initialize(in_name, attrs = [],
+                   required: false, desc: nil, default: nil, enums: [], regex: nil, aliases: [])
       self.name     = in_name
       self.advanced = false
       self.evasion  = false
-      self.required = attrs[0] || false
-      self.desc     = attrs[1]
-      self.default  = attrs[2]
-      self.enums    = [ *(attrs[3]) ].map { |x| x.to_s }
-      regex_temp    = attrs[4] || nil
+      self.aliases  = aliases
+
+      if attrs.is_a?(String) || attrs.length == 0
```

Comment:
Nitpick: Brixen once told me that :is_a? is costly as it hits up the GC or some other low level VM component. AFAIK, its more efficient to call :responds_to?(:upcase)

Reply:
Not seeing this in the synthetic case on ruby 2.3:
```
$ time ruby -e '10_000_000.times { "arg".is_a? String }

real    0m0.777s
user    0m0.774s
sys     0m0.004s

$ time ruby -e '10_000_000.times { "arg".respond_to? :upcase }

real    0m1.019s
user    0m1.007s
sys     0m0.014s
```